packages/radix-vue/src/PinInput/PinInputInput.vue

```diff
@@ -104,12 +104,18 @@ function handlePaste(event: ClipboardEvent) {
       continue
 
     input.value = value
-    context.modelValue.value[i] = value
+    updateModelValueAt(i, value)
     input.focus()
   }
   inputElements.value[lastIndex]?.focus()
 }
 
+function updateModelValueAt(index: number, value: string) {
+  const tempModelValue = context.modelValue.value
```

Comment:
Because `Array` is a "Reference value" in JS,  we need to assign the array differently, else the same issue still occur.

```ts
  const tempModelValue = [...context.modelValue.value]
```